Template categories + search

nikku opened the pull request:
_Replace options, including templates can now be optionally classified by category:_

![image](https://user-images.githubusercontent.com/58601/160406671-b11d3425-d61b-4f32-ae5c-a11a4d91ae7b.png)

Also, an additional _search_ meta-data allows to specify search terms not displayed through name, label or description.


----

Try it out via 

```
npx @bpmn-io/sr bpmn-io/bpmn-js-connectors-extension#template-categories-search -c 'npm start'
```

Enable _create/append anything modeling UX_ [via a feature toggle](http://localhost:8081/?aa=1).

pinussilvestrus commented on example/.camunda/element-templates/http-json-connector.json:
```diff
@@ -7,6 +7,10 @@
     "appliesTo": [
       "bpmn:ServiceTask"
     ],
+    "category": {
+      "id": "rest",
+      "name": "REST templates"
+    },
```
Will that live outside of the JSON Schema and only be used for the extension?

nikku replied:
No, that should be part of our official element template schema, if we decide to go down this route.

pinussilvestrus replied:
👍

nikku replied:
Opened an issue: https://github.com/bpmn-io/bpmn-js-properties-panel/issues/633.

nikku commented:
@currandwyer I'd love to make basic grouping available to our users. Please have a look at this change and highlight and feedback potential improvements.

![image](https://user-images.githubusercontent.com/58601/160344546-fb7d9b57-1e01-4598-b091-e1f21d4a2189.png)

![image](https://user-images.githubusercontent.com/58601/160344470-f60f1797-5d4c-4b26-bbbd-7f7985afc1e3.png)

currandwyer commented:
@nikku I think this looks great, particularly in this append anything context. I also expect that append anything is the direction of travel for our modeling UX, so this will be a nice feature to have. When this does happen, I suggest changing the title to "Append *specialized* element" or something like this, to be specific and precise about the purpose of the component.

I realize it's out of scope, but I think that adding the ability to filter would be nice to have alongside this. 

I don't think it makes sense to launch categories until we have a higher number of specialized tasks to categorize. Categorizing the REST and SendGrid connectors is a lipstick on a pig solution. Rather than categorizing them, we should focus on shipping auth so that the auth configuration exists within the properties panel, as it should. Then we only have two connectors, not enough to categorize. In any case, the categories will work differently (for more higher order distinctions), and we mustn't confuse these different mental models so early on.

I think it is very important not to refer to connectors as templates. We must align on one word (whether it is _connectors_ or _integrations_ or whatever) to refer to these things. The element template is a part of the connector, but it's not the main thing. Introducing this additional technical language, which requires users to develop an understanding of our technical architecture, could be unnecessarily confusing.

nikku commented:
Thanks for your comments, both of you.

> I realize it's out of scope, but I think that adding the ability to filter would be nice to have alongside this.

Agreed @currandwyer.

nikku commented:
Updated to not show the category, unless provided in `connectors` mode: 

![image](https://user-images.githubusercontent.com/58601/160394788-2b5de206-0792-40b1-b41c-f1b7c6b820f8.png)

----

Also updated the branch so it can be merged independently.
